src/java.base/share/classes/java/lang/Enum.java

```diff
@@ -166,13 +168,29 @@ public final boolean equals(Object other) {
         return this==other;
     }
 
+    /**
+     * The hash code of this enumeration constant.
+     *
+     * @implNote Once initialized, the field value does not change.
+     * Hotspot's identity hash code generation also never returns zero
+     * as the identity hash code. This allows to treat zero as the marker
+     * for the un-initialized value for both {#link @Stable} and the lazy
+     * initialization code.
+     */
+    @Stable
+    private int hash;
```

Comment:
Note that `transient` is not needed here, as per https://docs.oracle.com/en/java/javase/17/docs/specs/serialization/serial-arch.html#serialization-of-enum-constants

> The serialized form of an enum constant consists solely of its name; field values of the constant are not present in the form.